lib/ansible/modules/clustering/kubevirt/k8s_service.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_service
+
+short_description: Manage Services on Kubernetes
+
+description:
+    - Use Openshift Python SDK to manage Services on Kubernetes
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    state:
+        description:
+            - Determines if an object should be created, patched, or deleted. When set to C(present), an object will be created, if it does not already exist. If set to C(absent), an existing object will be deleted. If set to C(present), an existing object will be patched, if its attributes differ from those specified.
+        default: present
+        choices:
+            - present
+            - absent
+        type: str
+    name:
+        description:
+            - Use to specify a Service object name.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Use to specify a Service object namespace.
+        required: true
+        type: str
+    selector:
+        description:
+            - TODO
+        default: []
+        type: dict
+    type:
+        description:
+            - TODO
+        default: ClusterIP
+        choices:
+            - NodePort
+            - ClusterIP
+            - LoadBalancer
+    ports:
+        description:
+            - TODO
+        required: true
+        type: list
+    labels:
+        description:
+            - Labels are key/value pairs that are attached to objects.
+        default: {}
+        type: dict
+    spec:
+        description:
+            - Use to specify any additional parameters to be passed to the cluster. Everything defined here takes precedence over this module's other parameters.
+        default: {}
+        type: dict
+
+extends_documentation_fragment:
+  - k8s_auth_options
+  - k8s_resource_options
+
+requirements:
+  - python >= 2.7
+  - openshift >= 0.6.2
+'''
+
+EXAMPLES = '''
+TODO
+'''
+
+RETURN = '''
+service:
+    description:
+      - The Service object.
+    returned: success
+    type: complex
+'''
+
+import copy
+import traceback
+
+from collections import defaultdict
+
+from ansible.module_utils.k8s.common import AUTH_ARG_SPEC, COMMON_ARG_SPEC
+from ansible.module_utils.k8s.raw import KubernetesRawModule
+
+from openshift import watch
```

Comment:
I think you don't need this.

Reply:
That's a good general question. For something like PVC creation or replicaset modification it makes sense to have a --wait options, since volume space allocation/new instance creation might take time. But there's no way this makes sense for a Service module, right?

Reply:
With this comment I meant you don't use it now. But to your question, I am unsure how services works. the question is what it should wait for ... The IP address to appear ?

Reply:
The real question is whether we need the address at all.

Reply:
Yeah, I was hoping you'd know… ;) Anyway, as is tradition, I guess we can start with the initial version of this module not supporting --wait, and then figure out if we want/need it later.

Reply:
Users creates service just because of address, no? Or what's the use case of creating service without the need of address?

Reply:
ATM, I do not see a need for IP so I think there is no need to wait.

Reply:
Ah right, in case of nodeport. Ok, then. Thanks.

Reply:
Running the code as is results with the clusterIP getting assigned without having to wait for anything. So if one wants to use that info in the playbook, it's available as `result.spec.clusterIP`.